### PR TITLE
Frontend functions

### PR DESCRIPTION
**Pull request by BogdanKalysh**

### CODE REVIEW

**Review by Nasar-Mykytskyj — COMMENTED** [review]
norm

File: budget_manager_frontend/sources/mainWindow.cpp [hunk 1]
```diff
@@ -166,50 +128,88 @@ void MainWindow::finishedPostTransactions()
...
+{
+    qreal totalsum = 0;
 
-//void MainWindow::updatePiechart()
-//{
-//    series->clear();
-//    for(Category& cat: categories){
-//        series->append(getCategoryTotalSum(cat.getName()),cat.getName(),cat.getColor());
-//    }
+    for(Transaction &transac : transactions){
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/mainWindow.cpp:176]: it is better to use accumulate and lambda
https://en.cppreference.com/w/cpp/algorithm/accumulate

File: budget_manager_frontend/sources/mainWindow.cpp [hunk 2]
```diff
@@ -166,50 +128,88 @@ void MainWindow::finishedPostTransactions()
...
+            QPieSlice *slice = series->append(cat.getName(), getCategoryTotalSum(cat.getName()));
+            slice->setBorderWidth(0);
+            slice->setPen(Qt::NoPen);
+            slice->setColor(QColor(cat.getColor()));
+        }
+    }else{
+        QPieSlice *slice = series->append("0", 1);
+        slice->setBorderWidth(0);
+        slice->setPen(Qt::NoPen);
+        slice->setColor(QColor("#528bba"));
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/mainWindow.cpp:200]: Move to constant (something like DEFAULT_COLOR)

File: budget_manager_frontend/sources/mainWindow.cpp [hunk 3]
```diff
@@ -166,50 +128,88 @@ void MainWindow::finishedPostTransactions()
...
+    series->clear();
+    if(transactions.size() != 0){
+        for(Category& cat: categories){
+            amountSum += getCategoryTotalSum(cat.getName());
+            QPieSlice *slice = series->append(cat.getName(), getCategoryTotalSum(cat.getName()));
+            slice->setBorderWidth(0);
+            slice->setPen(Qt::NoPen);
+            slice->setColor(QColor(cat.getColor()));
+        }
+    }else{
```
Comment (AndriyFedorenko) [inline budget_manager_frontend/sources/mainWindow.cpp:196]: formatting

File: budget_manager_frontend/forms/loginwindow.ui [hunk 4]
```diff
@@ -196,7 +196,7 @@
           <string notr="true"/>
          </property>
          <property name="text">
-          <string/>
+          <string>bohdan@gmail.com</string>
```
[the remark this answers is not in the document]
Comment (BogdanKalysh) [reply budget_manager_frontend/forms/loginwindow.ui:199]: No we used it for debug, ill remove it

File: budget_manager_frontend/sources/mainWindow.cpp [hunk 5]
```diff
@@ -25,80 +21,46 @@ MainWindow::MainWindow(User user, QSharedPointer<QNetworkAccessManager> manager,
...
-//    QString getCategoriesQuery = "http://127.0.0.1:5000/rating/getcategories?u_email=" + user.getEmail();
-//    QNetworkReply *categoriesReply = manager->get(QNetworkRequest(QUrl(getCategoriesQuery)));
-//    connect(categoriesReply, &QNetworkReply::readyRead, this, &MainWindow::readCategories);
 
-//    QString getTransactionsQuery = "http://127.0.0.1:5000/rating/gettransactions?u_email=" + user.getEmail() + "&lim=10";
-//    QNetworkReply *transactionsReply = manager->get(QNetworkRequest(QUrl(getTransactionsQuery)));
-//    connect(transactionsReply, &QNetworkReply::readyRead, this, &MainWindow::readTransactions);
+void MainWindow::updateTransactions()
+{
+    QString getTransactionsQuery = "http://localhost:5000/transaction?" + jsonbuilder::USER_ID + "=" +
```
[the remark this answers is not in the document]
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/mainWindow.cpp:52]: Moved them into constants

File: budget_manager_frontend/sources/mainWindow.cpp [hunk 6]
```diff
@@ -166,50 +128,88 @@ void MainWindow::finishedPostTransactions()
...
+    chartView->chart()->addSeries(series);
+    chartView->setRenderHint(QPainter::Antialiasing);
+    chartView->chart()->legend()->setVisible(false);
+    chartView->chart()->setBackgroundVisible(false);
+
+    sumAmountLabel->setStyleSheet("font-size: 33px; color:white;");
+
+    QPixmap settingsPixMap(":/new/img/settings_icon.png");
+    QPixmap userPixMap(":/new/img/user_icon.png");
+    int w(40), h(40);
```
[the remark this answers is not in the document]
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/mainWindow.cpp:152]: added const

File: budget_manager_frontend/sources/mainWindow.cpp (hunk 2 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/mainWindow.cpp:200]: moved into constant

File: budget_manager_frontend/sources/mainWindow.cpp (hunk 3 above)
Comment (BogdanKalysh) [reply budget_manager_frontend/sources/mainWindow.cpp:196]: removed else

**Review by Nazar-TK — APPROVED** [review]

**Pull request merged**